### PR TITLE
Add titus-optimize dependency

### PR DESCRIPTION
**Pull request by gabrielhartmann**

### CODE REVIEW

File: tests/isolate/test_cpu.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ class TestCpu(unittest.TestCase):
 
     def test_assign_one_thread_empty_cpu(self):
         """
-        Workload 0: 1 cores --> p:0 c:0 t:0
+        Workload 0: 1 thread --> (p:0 c:0 t:0)
```
Comment (gabrielhartmann) [inline tests/isolate/test_cpu.py]: Why the change to `thread`?  This means `Workload 0` is requesting 1 thread, and it should get the thread 0 on core 0 on package 0.

File: titus_isolate/isolate/cpu.py [hunk 2]
```diff
@@ -1,11 +1,124 @@
...
+
 from titus_isolate.model.processor.utils import get_emptiest_core, is_cpu_full
 from titus_isolate.model.workload import Workload
 from titus_isolate.utils import get_logger
 
+from titus_optimize.compute import optimize_ip
+
 log = get_logger()
 
+def assign_threads(cpu, workload, workload_insertion_times={}):
```
Comment (gabrielhartmann) [inline titus_isolate/isolate/cpu.py]: Can you please document the `workload_insertion_times` parameter.  It's unclear to me what it's role is.

File: titus_isolate/isolate/cpu.py [hunk 3]
```diff
@@ -33,7 +147,7 @@ def assign_threads(cpu, workload):
     return claimed_threads + assign_threads(cpu, Workload(workload.get_id(), thread_count, workload.get_type()))
 
 
-def free_threads(cpu, workload_id):
+def free_threads_greedy(cpu, workload_id):
```
Comment (gabrielhartmann) [inline titus_isolate/isolate/cpu.py]: Dead code now.  Let's just remove the method.

File: titus_isolate/isolate/cpu.py (hunk 2 above)
Comment (rostyboost) [reply titus_isolate/isolate/cpu.py]: done

File: titus_isolate/isolate/cpu.py (hunk 3 above)
Comment (rostyboost) [reply titus_isolate/isolate/cpu.py]: As discussed in person, since we want the fallback I kept it.
